StablecoinResearch/DualTokenSim/source/Tokens/collateral_token.py

```
@@ -0,0 +1,71 @@
+
+        Raises:
+            TypeError: If algorithmic_stablecoin is not an instance of AlgorithmicStablecoin.
+            RuntimeError: If the stablecoin is already tied to another CollateralToken.
+        """
+        # Validate the algorithmic_stablecoin argument
+        if not isinstance(algorithmic_stablecoin, AlgorithmicStablecoin):
+            raise TypeError("algorithmic_stablecoin must be an instance of AlgorithmicStablecoin.")
+
+        # Check if the stablecoin is already tied to another collateral token
+        if algorithmic_stablecoin._tied:
+            raise RuntimeError("This stablecoin is already tied to another CollateralToken.")       
```

Ruff (0.12.2)

Avoid specifying long messages outside the exception class

(TRY003)